lib/rbs/writer.rb

```diff
@@ -232,10 +232,11 @@ def write_member(member)
     def method_name(name)
       s = name.to_s
 
-      if /\A#{Parser::KEYWORDS_RE}\z/.match?(s)
-        "`#{s}`"
-      else
+      if [:tOPERATOR, :kAMP, :kHAT, :kSTAR, :kLT, :kEXCLAMATION, :kSTAR2, :kBAR].include?(Parser::PUNCTS[s]) ||
+          (/\A[a-zA-Z_]\w*[?!=]?\z/.match?(s) && !/\A#{Parser::KEYWORDS_RE}\z/.match?(s))
```

Comment:
`RBS::Writer` escapes keywords, for example: ``def `def`: () -> untyped`. But this escaping is not necessary because it is allowed.

I keep this escaping because of compatibility, and I'm not sure why this escaping needs to keep. https://github.com/ruby/rbs/blob/a1e62612f2057a47259f4add6d15fe2700d6bf58/test/rbs/writer_test.rb#L114


If this escaping is not necessary, I can remove the escaping easily by removing `&& !/\A#{Parser::KEYWORDS_RE}\z/.match?(s)` condition.


What do you think? @soutaro

Reply:
It makes sense. I don't think we need to keep the quoting behavior. 

Let's make it:

1. Unquoted even if it is a keyword (proposed change)
2. Unquoted if it is safe `/\A[a-zA-Z_]\w*[?!=]?\z/`
3. Quoted otherwise

Reply:
Thanks for your comment. I applied this charge with https://github.com/ruby/rbs/pull/537/commits/2b032125d03453cc54b48c9d706d8cea03d5190b.

But I noticed we still need to quote `self` and `self?` as a method name because it causes syntax error without quote.
So I replace  `Parser::KEYWORDS_RE` with `/\Aself\??\z/`.